packages/dbt/sodasql/dbt.py

```diff
@@ -56,9 +57,11 @@ def parse_manifest(
     ------
     https://docs.getdbt.com/reference/artifacts/manifest-json
     """
+    logger = logging.getLogger(__name__)
+
     dbt_v4_schema = "https://schemas.getdbt.com/dbt/manifest/v4.json"
     if manifest["metadata"]["dbt_schema_version"] != dbt_v4_schema:
-        raise NotImplementedError("Dbt manifest parsing only supported for V4 schema.")
+        logger.warning("Dbt manifest parsing only tested for v4 schema.")
```

Comment:
parsing worked for `v3` too, did not add a proper test for that, tested it manually

Reply:
Dunno exactly, I tested it with a v4 schema, not sure if that came from dbt v1.

Reply:
Just so I understand a bit more and see if I can help.

Do I understand well that you made a new dbt project with dbt v1 and got your artifacts to be v4 by default?

Reply:
I don't know what version for dbt was used to create the artifacts that are found in the test folder. 

What is the current scenario: we test with a v4 artifact schema. 

The question is: are we going to test for all versions of the artifact schema? Or do we leave it like this and do a regression test once we find a schema for which the conversion breaks. Note: we can constrain the versions of the artifacts by setting the version of the dbt dependency e.g. `dbt>=1.0` and we use dbt internal data classes so that should help with the conversion - if the user uses the same dbt version

Reply:
I would enforce dbt >= 1 (I think that's already now the case in the `setup.py` file). 

I personally don't want to really do version checks and import checks unless we have to. v4 schemas have now changed:
- `CompiledTestNode` to `CompiledGenericTestNode`
- `ParsedTestNode` to `ParsedGenericTestNode`

if we support < v4 it means we have to import different classes from the  `contracts` module and although we will have to do it one day, the risk of it changing throughout the lifetime of dbt v1 is unlikely.

Reply:
What we could do is check that dbt v1 always generates v4 schemas and in that case we should be good.

Reply:
The dbt cloud we use for testing uses the dbt version 1.0.0. This returns a schema v4. I removed the checks, and we solve this by the `dbt` version constrain.